src/grammar/grammar.ts

```diff
@@ -547,7 +547,7 @@ semantics.addOperation<ASTNode>('resolve_expression', {
 
     // Literals
     integerLiteral(n) {
-        return createNode({ kind: 'number', value: BigInt(n.sourceString), ref: createRef(this) }); // Parses dec-based integer and hex-based integers
+        return createNode({ kind: 'number', value: BigInt(n.sourceString.replaceAll('_', '')), ref: createRef(this) }); // Parses dec, hex, and bin numbers
```

Comment:
Hmm, looks like we translate all integer literals into FunC's decimal literals (because we strip this info here). Let's not change it now, but might be nice to have direct correspondence (modulo binary and octal literals, which FunC to does not support yet)